### PR TITLE
SSCS-9714: Child Support - Update Other Party Data Event

### PR DESCRIPTION
**Pull request by rasikasri**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-9714

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -368,6 +368,7 @@ dependencies {
     //remove when secure doc store is live
     compile group: 'uk.gov.hmcts.reform', name: 'document-management-client', version: '7.0.0'
 
+    compile files('/Users/Rasika.Abeyrathna/Workspce/dev/hmcts/SSCS_Common/sscs-common/build/libs/sscs-common-DEV-SNAPSHOT.jar')
```
Comment (ryandac) [inline build.gradle:371]: remove

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/UpdateOtherPartyDataIt.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class UpdateOtherPartyDataIt extends AbstractEventIt {
+
+    @Before
+    public void setup() throws IOException {
+        setup("callback/updateOtherPartyCallback.json");
```
Comment (ryandac) [inline src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/UpdateOtherPartyDataIt.java:25]: 😭 Not everyone likes JSON. It's not _that_ many lines of code to create the appeal with other party data.
Ignore this comment, if you love this JSON file.

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+    private OtherPartyDataUtil() {
+    }
+
+    public static void updateOtherPartyUcb(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getOtherParties() != null) {
+            sscsCaseData.setOtherPartyUcb(sscsCaseData.getOtherParties().stream()
+                    .filter(o -> YesNo.isYes(o.getValue().getUnacceptableCustomerBehaviour()))
+                    .map(o -> o.getValue().getUnacceptableCustomerBehaviour().getValue())
+                    .findAny().orElse(YesNo.NO.getValue()));
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:22]: Use static imports as much as possible, for readability
Move the .orElse into a new line for the same reason.

File: build.gradle (hunk 1 above)
Comment (rasikasri) [reply build.gradle:371]: Done

File: src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/UpdateOtherPartyDataIt.java (hunk 2 above)
Comment (rasikasri) [reply src/IntegrationTests/java/uk/gov/hmcts/reform/sscs/callback/UpdateOtherPartyDataIt.java:25]: I'll keep this in mind for the next time

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java (hunk 3 above)
Comment (rasikasri) [reply src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:22]: Done

**Review by ryandac — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CcdValue;
+import uk.gov.hmcts.reform.sscs.ccd.domain.OtherParty;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+
+
+public class OtherPartyDataUtil {
```
Comment (gmikerlemos) [inline src/main/java/uk/gov/hmcts/reform/sscs/util/OtherPartyDataUtil.java:15]: Could this not be merged into PartiesOnCaseUtil or the other way around?

**Pull request merged**